src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -450,20 +465,29 @@ def engine_forward(
         # as such, a new context needs to be created since we are no longer in the
         # main thread. That is why `engine_` is prepended to each of the timer phase
         # names in this context
+
         with self.timer_manager.new_timer_context(total_inference=False) as timer:
             streamer = context.get("streamer")
 
             if not self.cache_support_enabled:
-                tokens, prompt_logits = self.multitoken_engine(engine_inputs)
-                return numpy.array([tokens]), prompt_logits
+                prompt_logits = self.multitoken_engine(engine_inputs)
+                token_generator = TokenGenerator(prompt_logits[0])
```

Comment:
```suggestion
                token_generator = TokenGenerator(prompt_logits)
```
maybe let the `TokenGenerator` take care of indexing.

Reply:
`TokenGenerator` should not take care of indexing. That means logits can be of any arbitrary shape. 

One change I can see is to just pass the length of the logit (51200) instead of the actual logit.

Reply:
It could take care of indexing quite nicely.
If logits.shape == 3 this always means (batch_size, sequence_len, vocab size)
elif logits.shape == 2 (sequence_len, vocab_size)
elif logit.shape == 1 (vocab_size)